src/Apps/Order/Routes/Shipping2/Hooks/useShippingContext.ts

```diff
@@ -0,0 +1,173 @@
+import { createContext, useContext, useMemo, useState } from "react"
+import { FormikProps } from "formik"
+import { ShippingProps, ShippingRouteStep } from "Apps/Order/Routes/Shipping2"
+import {
+  FulfillmentValues,
+  ShipValues,
+} from "Apps/Order/Routes/Shipping2/FulfillmentDetails"
+import {
+  addressWithFallbackValues,
+  getDefaultUserAddress,
+  FulfillmentType,
+} from "Apps/Order/Routes/Shipping2/Utils/shippingUtils"
+import { extractNodes } from "Utils/extractNodes"
+import {
+  ParsedOrderData,
+  useParseOrderData,
+} from "Apps/Order/Routes/Shipping2/Hooks/useParseOrderData"
+
+type FulfillmentHelpers = Pick<
+  FormikProps<FulfillmentValues>,
+  "handleSubmit" | "isValid" | "values"
+>
+export interface ShippingContextProps {
+  parsedOrderData: ParsedOrderData
+  initialValues: {
+    fulfillmentDetails: FulfillmentValues
+    shippingQuotes: {
+      selectedShippingQuoteId?: string | null
+    }
+  }
+  step: ShippingRouteStep
+  helpers: {
+    fulfillmentDetails: FulfillmentHelpers & {
+      setFulfillmentFormHelpers: (helpers: FulfillmentHelpers) => void
+    }
+  }
+}
+
+export const ShippingContext = createContext<ShippingContextProps>({} as any)
+
+export const useShippingContext = () => {
+  return useContext(ShippingContext)
+}
+
+// Get information from the order and user, parsedOrderData (shared, computed context about order)
+// initial values for forms, and the current step in the shipping route
+// values for forms
+export const useComputeShippingContext = (
+  props: ShippingProps
+): ShippingContextProps => {
+  const parsedOrderData = useParseOrderData(props)
+  const initialValues = getInitialValues(props, parsedOrderData)
+
+  const { isArtsyShipping } = parsedOrderData
+
+  const step: ShippingRouteStep = useMemo(() => {
+    if (isArtsyShipping) {
+      return "shipping_quotes"
+    }
+    return "fulfillment_details"
+  }, [isArtsyShipping])
+
+  return {
+    parsedOrderData,
+    initialValues,
+    step,
+    helpers: {
+      fulfillmentDetails: useFulfillmentDetailsHelpers(),
+    },
+  }
+}
+
+const useFulfillmentDetailsHelpers = (): ShippingContextProps["helpers"]["fulfillmentDetails"] => {
+  // Because there is a single button for both fulfillment details and
+  // shipping quote steps (and duplicated in the sidebar)
+  // we need to hack some formik values UP from the fulfillment details
+  // form.
+  // handleSubmit: Used to submit the form
+  // isValid: Used to disable the button
+  // values: Used to get the form values for un-saving the address if the user
+  //   unchecks it after saving it in the fulfillment details step.
```

Comment:
I'm getting nitpicky here, but this is conventional code quality standards stuff, so please bear with me.  

Please use paragraph breaks more often for detailed comments, if you want folks to read them. Everything is all smooshed together. You're describing something, then on the very next line, with no break in between, there's some kind of psudo doc-block style argument descriptions. Hit enter 🙏 And ask yourself if those argument comments are necessary here, versus putting them right above where the definition actually is. There it will be seen, and more useful. 

```
// Used to submit the form
handleSubmit: () => {},
// Used to disable the button
isValid: false 
```

Also, in JavaScript, there's multi-line comments, which are very useful. All you have to do is type `/**` and hit enter and VSCode will auto expand. For multi-line comments, please use multiline comments.

Lastly, per our VSCode config, there is a right margin guide that should be visible, set to 80 character width. When writing code comments, write them out to that guide. Everything feels so arbitrary which makes legibility hard to read. Prettier automatically hits the 80 character margin and formats, but with comments its the devs responsibility. Best just to get into the habit.

Reply:
Went through the main files and tried to reformat comments that were still doing something, remove others that went on too long, left a few in place that include old code references and need to be addressed in upcoming work. I'll try switch to multi-line when a comment busts through the first line in the future